model/impl/src/main/java/org/opennms/oce/model/impl/ModelImpl.java

```diff
@@ -68,6 +72,81 @@ public ModelObject getRoot() {
         return root;
     }
 
+    @Override
+    public void updateObjects(List<ModelObject> moList){}
+
+    @Override
+    public void updateObject(ModelObject mo){}
+
+    @Override
+    public void addObjects(List<ModelObject> moList) {
+
+        if(moList.isEmpty()) {
+            LOG.info("Loaded model objects list is empty.");
+            return;
+        }
+        else {
+            LOG.debug("Model object list size is : {}", moList.size());
+        }
+
+        /** simple iteration for a while
+         * THINGS TO CONSIDER:
+         * - If model objects in the list are somehow related to each other, eg parent child (device and card)
+         * -- define order of insertion/orchestration?
+         * -- can they be combined as single insertion?
+         * -- etc
+        **/
+        for(ModelObject mo : moList) {
+            addObject(mo);
+        }
+    }
+
+    /**
+     * The assumption is that model object contains all its hierarchy, for example if it is device,
+     * then it should have cards and ports, but if it is a new card, then there should be parent provided
+     * @param mo
+     */
+    @Override
+    public void addObject(ModelObject mo) {
+        addObject((ModelObjectImpl)mo);
```

Comment:
Why do we need to cast this to a `ModelObjectImpl`?

Reply:
This is leftover from old "yesterday's" implementation. After discussion here I moved functions to Model from ModelImpl. I will update Impl accordingly but the focus of this PR is to define API (Model) so engine can see it.